docs/docs/aztec/concepts/pxe/index.md

```diff
@@ -8,44 +8,101 @@ importance: 1
+        Circuits --Oracle--> PXE
+    end
+    PXE --Queries world-state (causes privacy leaks)--> Node
+    Wallet --Track tx state (may be handled via PXE)--> Node
+
+```
 
 ## Components
 
-### ACIR simulator
+### Transaction Simulator
+
+An application will prompt the users PXE to execute a transaction (e.g. execute X function, with Y arguments, from Z account). The application or the wallet may handle gas estimation.
+
+The ACIR (Abstract Circuit Intermediate Representation) simulator handles the execution of smart contract functions by simulating transactions. It generates the required data and inputs for these functions. You can find more details about how it works [here](./acir_simulator.md).
 
-The ACIR (Abstract Circuit Intermediate Representation) simulator handles the accurate execution of smart contract functions by simulating transactions. It generates the required data and inputs for these functions. You can find more details about how it works [here](./acir_simulator.md).
+### Proof Generation
+
+After simulation, the wallet calls `proveTx` on the PXE with all of the data generated during simulation and any [authentication witnesses](../accounts/authwit.md) (for allowing contracts to act on behalf of the users' account contract).
+
+Once proven, the wallet sends the transaction to the network and sends the transaction hash back to the application.
 
 ### Database
 
-The database stores transactional data and notes within the user's PXE. In the Aztec protocol, the database is implemented as a key-value database backed by LMDB. There is an interface ([GitHub](https://github.com/AztecProtocol/aztec-packages/blob/ca8b5d9dbff8d8062dbf1cb1bd39d93a4a636e86/yarn-project/pxe/src/database/pxe_database.ts)) for this PXE database that can be implemented in other ways, such as an in-memory database that can be used for testing.
+The database stores transactional data and notes within the user's PXE.
 
 The database stores various types of data, including:
 
-- **Notes**: Encrypted representations of assets.
-- **Deferred Notes**: Notes that are intended for a user but cannot yet be decoded due to the associated contract not being present in the database. When new contracts are deployed, there may be some time before it is accessible from the PXE database. When the PXE database is updated, deferred note are decoded.
-- **Authentication Witnesses**: Data used to approve others from executing transactions on your behalf.
+- **Notes**: Data representing users' private state. These are often stored on-chain, encrypted to a user. The PXE will parse on-chain data to find notes relevant for accounts that it is aware of.
+- **Deferred Notes**: Notes that are intended for a user but cannot yet be decoded due to the associated contract not being present in the database. When new contracts are deployed, there may be some time before it is accessible from the PXE database. When the PXE database is updated, deferred notes are decoded.
```

Comment:
These don't exist anymore since PXE doesn't know how to process logs. If a contract is not registered then it's notes won't be seen until it is.

Reply:
ok, deleting mention of deferred notes